lib/auth/grpcserver.go

```diff
@@ -840,6 +869,18 @@ func (g *GRPCServer) SetAccessRequestState(ctx context.Context, req *authpb.Requ
 }
 
 func (g *GRPCServer) SubmitAccessReview(ctx context.Context, review *types.AccessReviewSubmission) (*types.AccessRequestV3, error) {
+	// Prevent users from submitting access reviews with the "promoted" state.
+	// Promotion is only allowed by SubmitAccessReviewAllowPromotion API in Ent.
+	if review.Review.ProposedState.IsPromoted() {
+		return nil, trace.BadParameter("state promoted can be only set when promoting to access list")
```

Comment:
As a general rule, I'd put restrictions like this in `ServerWithRoles` not `GRPCServer`.  There's nothing grpc-specific about this restriction.

Reply:
Moved.